go/app/main.go

```diff
@@ -20,20 +25,132 @@ type Response struct {
 	Message string `json:"message"`
 }
 
+type Item struct {
+	Name     string `json:"name"`
+	Category string `json:"category"`
+	Image    string `json:"image"`
+}
+
 func root(c echo.Context) error {
 	res := Response{Message: "Hello, world!"}
 	return c.JSON(http.StatusOK, res)
 }
 
+func getFileContent(c echo.Context) (map[string][]Item, error) {
+	// Get data from items.json
+	f, err := os.OpenFile("items.json", os.O_RDONLY|os.O_CREATE, os.ModePerm)
+	if err != nil {
+		c.Logger().Errorf("err: %v", err)
+		return nil, err
+	}
+	defer f.Close()
+
+	bytes, err := io.ReadAll(f)
+	if err != nil {
+		c.Logger().Errorf("err: %v", err)
+		return nil, err
+	}
+	items := make(map[string][]Item)
+	if len(bytes) > 0 {
+		err = json.Unmarshal(bytes, &items)
+		if err != nil {
+			c.Logger().Errorf("err: %v", err)
+			return nil, err
+		}
+	}
+	return items, nil
+}
+
+func imageHash(image *multipart.FileHeader, c echo.Context) ([]byte, error) {
+	imagePath := path.Join(ImgDir, image.Filename)
+	imageFile, err := os.Open(imagePath)
+	if err != nil {
+		c.Logger().Errorf("err: %v", err)
+		return nil, err
+	}
+	defer imageFile.Close()
+
+	// Hashing image file
+	hash := sha256.New()
+	if _, err := io.Copy(hash, imageFile); err != nil {
+		c.Logger().Errorf("err: %v", err)
+		return nil, err
+	}
+	hashedImage := hash.Sum(nil)
+
+	c.Logger().Infof("Receive image: %x.jpg", hashedImage)
+	return hashedImage, nil
+}
+
+func writeItems(item Item, c echo.Context) error {
+	// Add item to item.json
+	items, err := getFileContent(c)
+	if err != nil {
+		c.Logger().Errorf("err: %v", err)
+		return err
+	}
+	items["items"] = append(items["items"], item)
+	jsonData, err := json.Marshal(items)
+	if err != nil {
+		c.Logger().Errorf("err: %v", err)
+		return err
+	}
+
+	f, err := os.OpenFile("items.json", os.O_WRONLY, os.ModePerm)
+	if err != nil {
+		c.Logger().Errorf("err: %v", err)
+		return err
+	}
+	f.Write(jsonData)
```

Comment:
こちらファイルのクローズをして欲しいですね。
```suggestion
　　 defer f.Close()
	f.Write(jsonData)
	
```